lib/Test/Net/RabbitMQ.pm

```diff
@@ -156,17 +158,23 @@ has queues => (
 );
 
 has delivery_tag => (
-    traits  => [ qw(Counter) ],
-    is      => 'ro',
-    isa     => 'Num',
-    default => 0,
-    handles => {
-        _inc_delivery_tag   => 'inc',
-        _dec_delivery_tag   => 'dec',
-        _reset_delivery_tag => 'reset',
-    },
+    is      => 'rw',
+    isa     => 'Math::UInt64',
+    default => sub { uint64(0) },
+    clearer => '_reset_delivery_tag',
+    writer  => '_set_delivery_tag',
```

Comment:
Isn't it better to access the attribute value through moose reader/writer in the _inc/_dec methods?

Is $self->{delivery_tag}++ better than using reader/writer in _inc_delivery_tag?

I admit this is a degenerate case, and probably the worst possible pedagogical example of the usefulness of this encapsulation, but there could be triggers, overrides, etc.